tests/test_client.py

```diff
@@ -286,3 +292,67 @@ def test_batch_wait(self, client):
         for pos, success in success_tasks:
             assert(tasks[pos].pk == success.pk)
             assert inference_schema(2, 0, 'golden retriever', 0.8) == success['data']
+
+
+class TestClientRetry(object):
```

Comment:
maybe use `unittest.mock` (on `requests.get`?) to simulate all cases to be tested:
- retry on status codes
- retry on exceptions inheriting `RequestException`
- no retry on blacklist exceptions
- no retry on exceptions not inheriting `RequestException`

Reply:
Most things have been tested